src/Nethermind/Nethermind.Core/Bloom.cs

```diff
@@ -157,35 +157,48 @@ public void Add(LogEntry[] logEntries, Bloom blockBloom = null)
             }
         }
         
-        public bool IsMatch(LogEntry logEntry)
+        public bool Matches(LogEntry logEntry)
         {
-            if (!IsMatch(logEntry.LoggersAddress.Bytes))
+            if (Matches(logEntry.LoggersAddress))
             {
-                return false;
-            }
-            
-            for (int topicIndex = 0; topicIndex < logEntry.Topics.Length; topicIndex++)
-            {
-                Keccak topic = logEntry.Topics[topicIndex];
-                if (!IsMatch(topic.Bytes))
+                for (int topicIndex = 0; topicIndex < logEntry.Topics.Length; topicIndex++)
                 {
-                    return false;
+                    if (!Matches(logEntry.Topics[topicIndex]))
+                    {
+                        return false;
+                    }
                 }
-            }
 
-            return true;
+                return true;
+            }
 
+            return false;
         }
+
+        public bool Matches(Address address) => Matches(address.Bytes);
+        
+        public bool Matches(Keccak topic) => Matches(topic.Bytes);
+        
+        public bool Matches(ref (int Index1, int Index2, int Index3) indexes) => _bits[indexes.Index1] && _bits[indexes.Index2] && _bits[indexes.Index3];
```

Comment:
since we use this Index triplet so often, maybe we can make into a struct like:
struct BloomExtract { int Index1, int Index2, int Index3} ?